### PR TITLE
Add the RHEL Linux stage

### PR DESCRIPTION
**Pull request by SukruthGrandhiNI**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).
hon/blob/master/CONTRIBUTING.md).

- [ ] I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.
Not required for this change  

- [ ] I've added tests applicable for this pull request
Not required for this change

### What does this Pull Request accomplish?
1. Add support for automated system testing on RHEL.
2. Also change the name from nimi-bot to nimibot as per the [PR](https://ni.visualstudio.com/DevCentral/_git/ni-central/pullrequest/220448) comments.
3. Skip few NI-DCPower tests please refer Issue #1667 for more details.
4. Fixed Tox related issues- by default it was using python 3.9 version for building package dependencies, so changed default version to python3.
5. Fixed the issues in NI-ModInst product, which was caused due to removing NI-DMM hardware simulation.

### List issues fixed by this Pull Request below, if any.
Added support for RHEL system testing

### What testing has been done?

Pipeline results [link](https://ni.visualstudio.com/Users/_build/results?buildId=1664555&view=logs&j=800ca5a1-1414-5cac-5ac5-fcd6baaee307)

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -70,7 +70,7 @@ NI-DCPower Python API Status
 +-------------------------------+--------------------------+
 | NI-DCPower (nidcpower)        |                          |
 +===============================+==========================+
-| Driver Version Tested Against | 21.0.0                   |
+| Driver Version Tested Against | 21.3.0                   |
```
Comment (marcoskirsch) [inline README.rst:73]: It seems like these changes belong to a different PR.
The old nimi-bot is still online and hasn't been updated.
@sbethur what do you think, should we do upgrade old nimi-bot first and make the "Tested Against" changes along with that in own PR?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -13,6 +13,11 @@ parameters:
   - nimodinst
   - nise
   - niswitch
+- name: rhel_module_names
+  type: object
+  default: # Order of the modules is based on test execution time, longest first.
+  - nidmm
+  - niswitch
```
Comment (marcoskirsch) [inline azure-pipelines.yml:20]: More than these two drivers are supported in RHEL.
Please create an issue to add the rest of the supported drivers:

1. NI-DCPower
1. NI-SCOPE
1. NI-TClk

File: azure-pipelines.yml [hunk 3]
```diff
@@ -36,25 +41,38 @@ stages:
   - stage: system_tests_win64
     dependsOn: []
     pool:
-      name: nimi-bot
+      name: nimibot
```
Comment (marcoskirsch) [inline azure-pipelines.yml:44]: renaming nimi-bot -> nimibot should be separate PRs.
PRs should be single purpose and focused.

File: src/nidcpower/system_tests/test_system_nidcpower.py [hunk 4]
```diff
@@ -333,6 +333,7 @@ def test_create_and_delete_advanced_sequence(session):
         session.active_advanced_sequence = sequence_name
 
 
+@pytest.mark.skip(reason="Please refer the Bug:1667")
```
Comment (marcoskirsch) [inline src/nidcpower/system_tests/test_system_nidcpower.py:336]: Since the fix is trivial, I'd rather see a PR for that (don't look at precise text) than a skip. Note that this failure is introduced by upgrading drivers which I argue should be done in its own separate PR.

But if you insist, then the proper way to reference issues is:

```suggestion
@pytest.mark.skip(reason="Workaround for issue #1667")
```

GitHub convention is  \#1667 is a link to an issue.

**Review by marcoskirsch — CHANGES_REQUESTED** [review]

File: README.rst (hunk 1 above)
Comment (SukruthGrandhiNI) [reply README.rst:73]: @marcoskirsch  old nimi-bot has been deleted, there was some confusion. So it is little more effort is required to test against old nimi-bot

**Comment by sbethur** [comment]
test this please

File: README.rst (hunk 1 above)
Comment (sbethur) [reply README.rst:73]: @SukruthGrandhiNI, Marcos is talking about "Jenkins nimi-bot", which I just triggered for this PR.

I agree that we should update the "Driver Version Tested Against" in a separate PR.

I can update "Jenkins nimi-bot" but "IaC nimi-bot" is very close to being ready (on both platforms) and I don't mind waiting for it to be done before updating the Readme. It's ok to under-report the tested versions for a few days.

File: src/nidcpower/system_tests/test_system_nidcpower.py (hunk 4 above)
Comment (sbethur) [reply src/nidcpower/system_tests/test_system_nidcpower.py:336]: I suggested to disable the test since there might be other tests we want to make more robust in this regard and I'd rather do that in a separate PR.

File: generated/nidigital/tox-system_tests.ini [hunk 5]
```diff
@@ -3,7 +3,7 @@
 # test suite on all supported python versions. To use it, "pip install tox"
 # and then run "tox -c tox-system_tests.ini" from the driver directory. (generated/nidigital)
 [tox]
-envlist = py39-nidigital-wheel_dep,py{36,37,38,39}-nidigital-system_tests, py39-nidigital-coverage
+envlist = py3-nidigital-wheel_dep,py{36,37,38,39}-nidigital-system_tests, py39-nidigital-coverage
```
Comment (sbethur) [inline generated/nidigital/tox-system_tests.ini:6]: These tox ini files in "generated" folder are codegen'd from https://github.com/ni/nimi-python/blob/master/build/templates/tox-system_tests.ini.mako. You need to update that and run the code generator.

File: azure-pipelines.yml (hunk 3 above)
Comment (SukruthGrandhiNI) [reply azure-pipelines.yml:44]: Sorry @marcoskirsch  I cannot make this as a separate change.

Since new agents are created in  "nimibot", I will not able to test this pipeline changes if I make it as a separate change.

Comment (marcoskirsch) [reply azure-pipelines.yml:44]: ok

File: generated/nidigital/tox-system_tests.ini (hunk 5 above)
Comment (SukruthGrandhiNI) [reply generated/nidigital/tox-system_tests.ini:6]: Done

File: azure-pipelines.yml (hunk 2 above)
Comment (SukruthGrandhiNI) [reply azure-pipelines.yml:20]: https://github.com/ni/nimi-python/issues/1669

**Comment by SukruthGrandhiNI** [comment]
@marcoskirsch  and @sbethur need help in fixing the failed tests:  https://ni.visualstudio.com/DevCentral/_build/results?buildId=1672193&view=results.

These tests run as part of submission of changes.

**Review by marcoskirsch — APPROVED** [review]

**Review by sbethur — APPROVED** [review]
